app/workers/facilities/pssg_download.rb

```diff
@@ -22,16 +25,17 @@ def create_and_save_drive_time_data(drive_time_data)
       return if facility.nil?
 
       begin
-        drive_time_band = facility.drivetime_bands.find_or_initialize_by(vha_facility_id: id)
+        name = attributes&.dig('Name')
+        drive_time_band = facility.drivetime_bands.find_or_initialize_by(vha_facility_id: id, name: name)
         drive_time_band.unit = 'minutes'
         drive_time_band.min = attributes&.dig('FromBreak')
         drive_time_band.max = attributes&.dig('ToBreak')
-        drive_time_band.name = attributes&.dig('Name')
+        drive_time_band.name = name
         drive_time_band.polygon = extract_polygon(drive_time_data)
         drive_time_band.save
         facility.save
       rescue => e
-        logger.error e.message
+        raise PSSGDownloadError, e.message
```

Comment:
I think this begin/rescue should be around the `download_data` line in the `perform` method.